### PR TITLE
[CALCITE] Added support for INSERT with omitted values

### PR DESCRIPTION
**Pull request by michaelvdow**
Added support for omitted values in an INSERT statement:
`INSERT INTO <table_name> [VALUES] \(<value>?(,<value>?)*\);`

### CODE REVIEW

File: core/src/main/codegen/templates/Parser.jj [hunk 1]
```diff
@@ -798,6 +798,11 @@ SqlNodeList ParenthesizedQueryOrCommaListWithDefault(
         e = OrderedQueryOrExpr(firstExprContext)
     |
         e = Default()
+<#if parser.includeInsertWithOmittedValues>
+    |
+        LOOKAHEAD({ getToken(1).kind != RPAREN })
```
Comment (dstekol) [inline core/src/main/codegen/templates/Parser.jj:803]: Maybe you could add a comment clarifying the LOOKAHEAD statement here?

File: babel/src/main/codegen/includes/parserImpls.ftl [hunk 2]
```diff
@@ -1039,10 +1039,24 @@ SqlNode LiteralRowConstructor() :
 }
 {
     <LPAREN>
-    e = AtomicRowExpression() { valueList.add(e); }
     (
-        LOOKAHEAD(2)
-        <COMMA> e = AtomicRowExpression() { valueList.add(e); }
+        e = AtomicRowExpression()
```
Comment (dasch-1) [inline babel/src/main/codegen/includes/parserImpls.ftl:1043]: nit: Would it be worth breaking out this parsing logic into a helper function that's called in the initial case as well as in the repeated case below?

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 3]
```diff
@@ -1989,4 +1989,32 @@ private void checkParseInfixCast(String sqlType) {
...
+  }
+
+  @Test public void testInsertAllOmittedValues() {
+    final String sql = "insert into foo values (,,)";
+    final String expected = "INSERT INTO `FOO`\n"
+        + "VALUES (ROW(NULL, NULL, NULL))";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testInsertOneOmittedValueNoValueKeyword() {
```
Comment (dasch-1) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:2007]: nit: change to testInsertOneOmittedValueNoValuesKeyword

File: core/src/main/codegen/templates/Parser.jj (hunk 1 above)
Comment (michaelvdow) [reply core/src/main/codegen/templates/Parser.jj:803]: Yeah that's a good point, I'll add one here

File: babel/src/main/codegen/includes/parserImpls.ftl (hunk 2 above)
Comment (michaelvdow) [reply babel/src/main/codegen/includes/parserImpls.ftl:1043]: Yeah I think it'd be better to break it down like that

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java (hunk 3 above)
Comment (michaelvdow) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java]: Updated

**Pull request merged**